src/nbl/video/utilities/ImageRegionIterator.cpp

```diff
@@ -92,7 +92,7 @@ ImageRegionIterator::ImageRegionIterator(
 
         uint64_t offsetInCPUBuffer = region.bufferOffset;
         uint8_t* inCpuBufferPointer = const_cast<uint8_t*>(reinterpret_cast<const uint8_t*>(srcData) + offsetInCPUBuffer);
-        core::smart_refctd_ptr<asset::ICPUBuffer> inCPUBuffer = core::make_smart_refctd_ptr< asset::CCustomAllocatorCPUBuffer<core::null_allocator<uint8_t>, true> >(0xdeadbeefBADC0FFEull, inCpuBufferPointer, core::adopt_memory);
+        core::smart_refctd_ptr<asset::ICPUBuffer> inCPUBuffer = asset::ICPUBuffer::create({ 0xdeadbeefBADC0FFEull, inCpuBufferPointer }, core::adopt_memory);
```

Comment:
again wrong replacement of memory resource, this will crash because it will try to free